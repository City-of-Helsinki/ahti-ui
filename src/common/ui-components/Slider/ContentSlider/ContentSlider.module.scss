.title {
  text-shadow: var(--hds-ui-color-black) 1px 0 10px;
}
