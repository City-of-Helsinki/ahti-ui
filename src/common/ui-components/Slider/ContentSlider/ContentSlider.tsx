import React from 'react';
import Slider from 'react-slick';
import { useTranslation } from 'react-i18next';
import classNames from 'classnames';

import '../Slider.scss';
import { CommonSliderProps } from '../Slider';
import { Filter } from '../../../../../alltypes';
import Slide from '../Slide/Slide';

export type ContentSliderItem = {
  readonly categoryFilters: Filter[];
  readonly tagFilters: Filter[];
  readonly title: string;
  readonly imageUrl: string;
};

export interface ContentSliderProps extends CommonSliderProps {
  readonly className?: string;
  readonly items: ContentSliderItem[];
  readonly translated?: boolean;
  onClick?(item: ContentSliderItem): void;
}

const ContentSlider: React.FC<ContentSliderProps> = ({
  className,
  slideClassName,
  items,
  translated,
  onClick,
  infinite = false,
  initialSlide = 0,
  slidesToScroll = 1,
  slidesToShow = 2,
  speed = 500,
}) => {
  const sliderSettings = {
    arrows: true,
    initialSlide: initialSlide,
    infinite: infinite,
    speed: speed,
    slidesToShow: slidesToShow,
    slidesToScroll: slidesToScroll,
  };

  const { t } = useTranslation();

  return (
    <div className={classNames(className, 'commonSlider')}>
      <Slider {...sliderSettings}>
        {items.map((item: ContentSliderItem, id: number) => (
          <Slide
            key={id}
            imageUrl={item.imageUrl}
            className={slideClassName}
            onClick={() => onClick && onClick(item)}
          >
            <p>{translated ? t(item.title) : item.title}</p>
          </Slide>
        ))}
      </Slider>
    </div>
  );
};

export default ContentSlider;
