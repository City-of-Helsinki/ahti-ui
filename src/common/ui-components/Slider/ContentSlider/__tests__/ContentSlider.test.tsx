import React from 'react';
import { shallow } from 'enzyme';

import ContentSlider from '../ContentSlider';

describe('ContentSlider', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ContentSlider items={[]} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
