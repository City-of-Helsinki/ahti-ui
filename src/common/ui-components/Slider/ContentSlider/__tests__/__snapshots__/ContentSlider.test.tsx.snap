// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentSlider renders without crashing 1`] = `"<div class=\\"commonSlider\\"><div class=\\"slick-slider slick-initialized\\"><div class=\\"slick-list\\"><div class=\\"slick-track\\" style=\\"width:0%;left:NaN%\\"></div></div></div></div>"`;
